Trying to use

I'm sure I'm doing something wrong but when I try to do a make infrastructure I get the following:

/aws-openshift/terraform-aws-openshift>make infrastructure
# Get the modules, create the infrastructure.
terraform init && terraform get && terraform apply -auto-approve
Initializing modules...

Initializing the backend...

Initializing provider plugins...

The following providers do not have any version constraints in configuration,
so the latest version was installed.

To prevent automatic upgrades to new major versions that may contain breaking
changes, it is recommended to add version = "..." constraints to the
corresponding provider blocks in configuration, with the constraint strings
suggested below.

* provider.aws: version = "~> 2.19"
* provider.local: version = "~> 1.3"
* provider.template: version = "~> 2.1"

Terraform has been successfully initialized!

You may now begin working with Terraform. Try running "terraform plan" to see
any changes that are required for your infrastructure. All Terraform commands
should now work.

If you ever set or change modules or backend configuration for Terraform,
rerun this command to reinitialize your working directory. If you forget, other
commands will detect it and remind you to do so if necessary.

Error: Unsupported block type

  on modules/openshift/06-nodes.tf line 10, in data "template_file" "setup-master":
  10:   vars {

Blocks of type "vars" are not expected here. Did you mean to define argument
"vars"? If so, use the equals sign to assign it a value.


Error: Unsupported block type

  on modules/openshift/06-nodes.tf line 64, in data "template_file" "setup-node":
  64:   vars {

Blocks of type "vars" are not expected here. Did you mean to define argument
"vars"? If so, use the equals sign to assign it a value.


Error: Unsupported block type

  on modules/openshift/07-dns.tf line 11, in resource "aws_route53_zone" "internal":
  11:   tags {

Blocks of type "tags" are not expected here. Did you mean to define argument
"tags"? If so, use the equals sign to assign it a value.


Error: Unsupported block type

  on modules/openshift/09-inventory.tf line 5, in data "template_file" "inventory":
   5:   vars {

Blocks of type "vars" are not expected here. Did you mean to define argument
"vars"? If so, use the equals sign to assign it a value.

make: *** [infrastructure] Error 1

Any help would be appreciated.

--- a/README.md
+++ b/README.md
@@ -51,7 +51,7 @@ created, which is used to install the OpenShift Origin platform on the hosts.
 
 You need:
 
-1. [Terraform](https://www.terraform.io/intro/getting-started/install.html) - `brew update && brew install terraform`
+1. [Terraform (0.12 or greater)](https://www.terraform.io/intro/getting-started/install.html) - `brew update && brew install terraform`
 2. An AWS account, configured with the cli locally -
 ```
 if [[ "$unamestr" == 'Linux' ]]; then

--- a/main.tf
+++ b/main.tf
@@ -1,6 +1,15 @@
-//  Setup the core provider information.
+# Setup our providers so that we have deterministic dependecy resolution. 
 provider "aws" {
   region  = "${var.region}"
+  version = "~> 2.19"
+}
+
+provider "local" {
+  version = "~> 1.3"
+}
+
+provider "template" {
+  version = "~> 2.1"
 }
 
 //  Create the OpenShift cluster using our module.

--- a/modules/openshift/06-nodes.tf
+++ b/modules/openshift/06-nodes.tf
@@ -7,7 +7,7 @@ resource "aws_key_pair" "keypair" {
 //  Create the master userdata script.
 data "template_file" "setup-master" {
   template = "${file("${path.module}/files/setup-master.sh")}"
-  vars {
+  vars = {
     availability_zone = "${data.aws_availability_zones.azs.names[0]}"
   }
 }
@@ -61,7 +61,7 @@ resource "aws_instance" "master" {
 //  Create the node userdata script.
 data "template_file" "setup-node" {
   template = "${file("${path.module}/files/setup-node.sh")}"
-  vars {
+  vars = {
     availability_zone = "${data.aws_availability_zones.azs.names[0]}"
   }
 }

--- a/modules/openshift/07-dns.tf
+++ b/modules/openshift/07-dns.tf
@@ -8,7 +8,7 @@ resource "aws_route53_zone" "internal" {
   vpc {
     vpc_id = "${aws_vpc.openshift.id}"
   }
-  tags {
+  tags = {
     Name    = "OpenShift Internal DNS"
     Project = "openshift"
   }

--- a/modules/openshift/09-inventory.tf
+++ b/modules/openshift/09-inventory.tf
@@ -2,7 +2,7 @@
 //  inventory from the inventory template.
 data "template_file" "inventory" {
   template = "${file("${path.cwd}/inventory.template.cfg")}"
-  vars {
+  vars = {
     access_key = "${aws_iam_access_key.openshift-aws-user.id}"
     secret_key = "${aws_iam_access_key.openshift-aws-user.secret}"
     public_hostname = "${aws_eip.master_eip.public_ip}.xip.io"

